pkg/controller/datavolume-controller.go

```diff
@@ -425,92 +433,30 @@ func (r *DatavolumeReconciler) Reconcile(_ context.Context, req reconcile.Reques
 		}
 	}
 
-	if cloneStrategy == SmartClone {
-		return r.reconcileSmartCloneForExistingPvc(log, datavolume, pvc, pvcSpec, transferName)
-	}
-	if cloneStrategy == CsiClone {
-		// Right not a NO-OP so falls back to Host Assisted - implementation in progress
-	}
-	// Finally, we update the status block of the DataVolume resource to reflect the
-	// current state of the world
-	return r.reconcileDataVolumeStatus(datavolume, pvc)
-}
-
-func (r *DatavolumeReconciler) isCSIClonePossible(dataVolume *cdiv1.DataVolume, targetStorageSpec *corev1.PersistentVolumeClaimSpec) (bool, error) {
-	log := r.log.WithName("CSIClonePossible").V(3)
-
-	sourcePvcSpec := dataVolume.Spec.Source.PVC
-	if sourcePvcSpec == nil {
-		log.Info("no source PVC provided")
-		return false, nil
-	}
-	sourcePvcNs := sourcePvcSpec.Namespace
-	if sourcePvcNs == "" {
-		sourcePvcNs = dataVolume.Namespace
-	}
-	sourcePvc := &corev1.PersistentVolumeClaim{}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: sourcePvcNs, Name: sourcePvcSpec.Name}, sourcePvc); err != nil {
-		if k8serrors.IsNotFound(err) {
-			r.log.V(3).Info("Source PVC is missing", "source namespace", sourcePvcSpec.Namespace, "source name", sourcePvcSpec.Name)
-			return false, errors.New("source PVC not found")
+	if selectedCloneStrategy == SmartClone || selectedCloneStrategy == CsiClone {
+		if selectedCloneStrategy == CsiClone {
+			if pvc.Status.Phase == corev1.ClaimBound {
+				if err := r.setCloneOfOnPvc(pvc); err != nil {
+					return reconcile.Result{}, err
+				}
+			} else if pvc.Status.Phase == corev1.ClaimPending {
+				//not yet ready
+				return reconcile.Result{Requeue: true}, nil
+			}
+			// TODO: is there any work for ClaimLost,
```

Comment:
Do you think it is ok to do:
```
phase = cdiv1.Failed, 
eventType: corev1.EventTypeWarning,
reason:    ErrClaimLost,
```

When you see ClaimLost, then it is probably better to investigate, delete DV (to cleanup any transfer objects etc...), fix what is wrong and retry.

Reply:
I've seen cases become unlost so as long as "failed" is not terminal

Reply:
I can in fact give an example of becoming 'unlost'. Create a manual PV and PVC, and ensure they become bound. Then 'export' the yaml for both. Then delete the PV and PVC. Then create both PV and PVC again, the PVC will be lost for a little bit until it gets matched back with the PV.

Reply:
ok, so do you suggest a new phase - like  CloneTargetPVCLost, or leaving the Failed and just testing the scenario to make sure it can eventually continue if PVC changed from ClaimLost to Bound?

Reply:
Updated one condition and according to new test the CSI clone logic can continue when PVC is recovered from ClaimLost

Reply:
https://github.com/kubevirt/containerized-data-importer/blob/b04a47574e3e58a322d22bfc5629b55a2bfbc314/pkg/controller/datavolume-controller_test.go#L1195